src/js/components/Calendar/Calendar.js

```diff
@@ -474,6 +474,23 @@ const Calendar = forwardRef(
             <StyledDay sizeProp={size} fillContainer={fill} />
           </StyledDayContainer>,
         );
+      } else if (
+        /* Do not show adjacent days in 6th row if all days 
+        fall in the next month */
+        showAdjacentDays === 'trim' &&
+        otherMonth &&
+        weeks.length === 5 &&
+        day.getDate() >= days.length + 1
```

Comment:
Just trying to understand the logic here -- I'm not exactly following this last comparison. Could you elaborate?

Reply:
Good question.

So, the loop iterates through a range of dates, building up a week's worth of days in a `days` array. Whenever, the day is the first day of a week, it pushes that week of days into a `weeks` array and then resets `days` to be an empty array.

This condition comes into play specifically when building up the 6th week (i.e. last calendar row).

In the 6th week we know that 2 scenarios could occur:
- a) The beginning of the row will be populated with the end dates of the current month and the remaining days of the row filled in by dates in the next month (we want to populate row 6 here)
- b) The current month's dates were exhausted in row 5 and the minimum value for the first day in row 6 is the 1st of next month (we do not want to populate row 6 here)

If scenario **a)**,  by the time the `otherMonth` condition is satisfied the `days` will be populated with at least one date of the current month and the array length will be greater than 1 (i.e. `getDate()` of the first day of the next month) and the condition will not be satisfied.

If scenario **b)**, the `otherMonth` condition is satisfied on the first pass when days.length will be 0, and the `getDate()` will be greater or equal to 1 (i.e. the first of the next month). In this case, the condition is met and we add blank placeholder days instead of "visible" days.

Phew!!! That was a lot!

Hopefully you were able to follow that, if not, I can try again. 😬

Reply:
Gotcha, thanks for the detailed explanation. It took me better understanding how the `days` and `weeks` arrays are built up to grasp the logic fully, but now it makes sense. I think that will be the case for others that may come into this file in the future but hoping the comment that you are adding will help serve as a high-level explanation.

Reply:
@taysea I included the comment we discussed. Appreciate your feedback and suggestions.